src/thread_task.rs

```diff
@@ -514,16 +536,15 @@ pub(crate) unsafe fn rav1d_task_delayed_fg(
     in_0: *const Rav1dPicture,
 ) {
     let ttd: *mut TaskThreadData = &mut (*c).task_thread;
-    (*ttd).delayed_fg.in_0 = in_0;
-    (*ttd).delayed_fg.out = out;
-    (*ttd).delayed_fg.type_0 = RAV1D_TASK_TYPE_FG_PREP;
-    *(&mut *((*ttd).delayed_fg.progress).as_mut_ptr().offset(0) as *mut atomic_int) = 0 as c_int;
-    *(&mut *((*ttd).delayed_fg.progress).as_mut_ptr().offset(1) as *mut atomic_int) = 0 as c_int;
-    pthread_mutex_lock(&mut (*ttd).lock);
-    (*ttd).delayed_fg.exec = 1 as c_int;
-    pthread_cond_signal(&mut (*ttd).cond);
-    pthread_cond_wait(&mut (*ttd).delayed_fg.cond, &mut (*ttd).lock);
-    pthread_mutex_unlock(&mut (*ttd).lock);
+    let mut ttd_locked = (*ttd).lock.lock().unwrap();
+    ttd_locked.delayed_fg.in_0 = in_0;
+    ttd_locked.delayed_fg.out = out;
+    ttd_locked.delayed_fg.type_0 = RAV1D_TASK_TYPE_FG_PREP;
+    *(&mut *((*ttd).delayed_fg_progress).as_mut_ptr().offset(0) as *mut atomic_int) = 0 as c_int;
+    *(&mut *((*ttd).delayed_fg_progress).as_mut_ptr().offset(1) as *mut atomic_int) = 0 as c_int;
+    ttd_locked.delayed_fg.exec = 1 as c_int;
+    (*ttd).cond.notify_one();
+    drop((*ttd).delayed_fg_cond.wait(ttd_locked).unwrap());
```

Comment:
We don't need the `drop` since it's at the end of the `fn`, right?

Reply:
We need to use the return value of `wait`, and I figure explicitly dropping it makes that clear we are immediately unlocking (which is an unusual pattern).

Reply:
Can you use `let _ = ` instead?  That's the usual pattern for drop immediately.